operator/watchers/k8s_service_sync.go

```diff
@@ -99,7 +98,7 @@ type ServiceSyncConfiguration interface {
 // 'shared' specifies whether only shared services are synchronized. If 'false' then all services
 // will be synchronized. For clustermesh we only need to synchronize shared services, while for
 // VM support we need to sync all the services.
-func StartSynchronizingServices(ctx context.Context, clientset k8sClient.Clientset, services resource.Resource[*slim_corev1.Service], shared bool, cfg ServiceSyncConfiguration) {
+func StartSynchronizingServices(ctx context.Context, wg *sync.WaitGroup, clientset k8sClient.Clientset, services resource.Resource[*slim_corev1.Service], shared bool, cfg ServiceSyncConfiguration) {
```

Comment:
nit: It'd be nice to have a comment about the purpose of the waitgroup here. As far as I follow, it's about signalling to the caller when all cleanup has completed and the service handling has stopped. Until we address the longer term planned changes, this could be a bit confusing to follow without some more indications of the purpose like this.